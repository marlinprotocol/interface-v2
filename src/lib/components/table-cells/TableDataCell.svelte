<script lang="ts">
	import { tableClasses } from '$lib/atoms/componentClasses';
	import { cn } from '$lib/utils/helpers/commonHelper';
	export let styleClass = '';
</script>

<td class={cn(tableClasses.row, styleClass)} data-testid="table-data-cell">
	<slot />
</td>
