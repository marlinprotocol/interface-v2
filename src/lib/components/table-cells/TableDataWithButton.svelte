<script lang="ts">
	import { tableClasses } from '$lib/atoms/componentClasses';
	import { cn } from '$lib/utils/helpers/commonHelper';
	export let styleClass = '';
	export let firstRow: boolean = false;
</script>

<td class={cn(styleClass, firstRow ? 'pl-4' : '')} data-testid="table-data-with-button">
	<div class={tableClasses.row}>
		<div class="m-auto flex h-[35px] items-center">
			<slot name="line1" />
		</div>
	</div>
</td>
