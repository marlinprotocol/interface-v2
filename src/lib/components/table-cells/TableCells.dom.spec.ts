import { cleanup, render } from '@testing-library/svelte';
import TableDataCell from './TableDataCell.svelte';
import html from '@playpilot/svelte-htm';
import TableDataWithButton from './TableDataWithButton.svelte';

describe('Table Cells', () => {
	afterEach(() => {
		cleanup();
		vi.resetAllMocks();
	});

	it('renders TableDataCell successfully', () => {
		const { getByTestId, container } = render(TableDataCell, {
			styleClass: 'table-data-test-class'
		});
		expect(getByTestId('table-data-cell')).toBeTruthy();
		expect(getByTestId('table-data-cell').classList.contains('table-data-test-class')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	it('renders with slot ', () => {
		const { getByTestId, getByLabelText } = render(
			html`<${TableDataCell}><div aria-label="test-component-chidren">Test Component Chidren</div></${TableDataCell}>`
		);
		expect(getByTestId('table-data-cell')).toBeTruthy();
		expect(
			getByTestId('table-data-cell').contains(getByLabelText('test-component-chidren'))
		).toBeTruthy();

		expect(
			getByLabelText('test-component-chidren').innerHTML === 'Test Component Chidren'
		).toBeTruthy();
	});

	it('renders TableDataWithButton successfully', () => {
		const { getByTestId, container } = render(TableDataWithButton, {
			styleClass: 'table-data-with-button-test-class'
		});
		expect(getByTestId('table-data-with-button')).toBeTruthy();
		expect(
			getByTestId('table-data-with-button').classList.contains('table-data-with-button-test-class')
		).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	it('renders TableDataWithButton  with slots', () => {
		const { getByTestId, findByText } = render(
			html`<${TableDataWithButton}>
            <button slot='line1'>Test Component First Chidren</button>
            <button slot='line2'>Test Component Second Chidren</button>
            </${TableDataWithButton}>`
		);
		expect(getByTestId('table-data-with-button')).toBeTruthy();
		expect(findByText('Test Component First Chidren')).toBeTruthy();
		expect(findByText('Test Component Second Chidren')).toBeTruthy();
	});
});
