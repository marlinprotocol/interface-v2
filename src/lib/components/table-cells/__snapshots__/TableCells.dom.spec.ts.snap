// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table Cells > renders TableDataCell successfully 1`] = `
<body>
  <div>
    <td
      class="table-data-test-class font-normal text-sm py-3 whitespace-nowrap"
      data-testid="table-data-cell"
    />
    <!--&lt;TableDataCell&gt;-->
  </div>
</body>
`;

exports[`Table Cells > renders TableDataWithButton successfully 1`] = `
<body>
  <div>
    <td
      class="pb-8 pt-6 table-data-with-button-test-class font-semibold text-base py-6 whitespace-nowrap"
      data-testid="table-data-with-button"
    >
      <div
        class="flex flex-col items-center justify-start"
      >
        <div
          class="m-auto flex h-[35px] items-center"
        />
         
        <div
          class="flex h-[18px] items-center pt-[10px]"
        />
      </div>
    </td>
    <!--&lt;TableDataWithButton&gt;-->
  </div>
</body>
`;
